components/raftstore/src/store/peer.rs

```diff
@@ -2322,6 +2336,18 @@ where
                             "region_id" => self.region_id,
                         );
                     }
+                    // After the leadership changed, send `CasualMessage::Campaign` to the target
+                    // peer to campaign leader if there exists uncampaigned regions. It's used to
+                    // ensure that a leader is elected promptly for the newly created Raft
+                    // group, minimizing availability impact (e.g. #12410 and #17602.).
+                    if let Some(new_regions) = &self.uncampaigned_new_regions {
```

Comment:
What if it's just a normal transfer leader after split? The newly created region would already campaign a leader, while here followers trigger another campaign.

Reply:
For Leader, the `uncampaigned_list.empty() == true`; for Followers, the `uncampaigned_list` will be cleared automatically.

Reply:
But the transfer leader may be performed before the `uncampaigned_list` is cleared automatically on followers.